cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/pipeline/AppDeploymentInfo.java

```diff
@@ -72,13 +78,17 @@ public static Builder copyFrom(AppDeploymentInfo other) {
       .setConfigString(other.configString)
       .setOwnerPrincipal(other.ownerPrincipal)
       .setUpdateSchedules(other.updateSchedules)
-      .setRuntimeInfo(other.runtimeInfo);
+      .setRuntimeInfo(other.runtimeInfo)
+      .setChangeSummary(other.changeSummary)
+      .setAuthor(other.author)
+      .setParentVersion(other.parentVersion);
   }
 
   private AppDeploymentInfo(ArtifactId artifactId, Location artifactLocation, NamespaceId namespaceId,
                             ApplicationClass applicationClass, @Nullable String appName, @Nullable String appVersion,
                             @Nullable String configString, @Nullable KerberosPrincipalId ownerPrincipal,
-                            boolean updateSchedules, @Nullable AppDeploymentRuntimeInfo runtimeInfo) {
+                            boolean updateSchedules, @Nullable AppDeploymentRuntimeInfo runtimeInfo,
+                            @Nullable String changeSummary, @Nullable String author, @Nullable String parentVersion) {
```

Comment:
Can we take a `ChangeSummary` instead of multiple fields? Also, I think the parentVersion can be part of the ChangeSummary as well.

Reply:
author however is not part of the request, so we will have that separate